### PR TITLE
kvstore: Remove Consul support

### PR DESCRIPTION
**Pull request by pchaigno**
I was looking at a CI issue that involved Consul... and figured it's probably best to just finally remove the Consul support.

Consul support has been deprecated in v1.12 with commit fb65f8c55b ("docs: Deprecate Consul support"). It's been almost three years and it seems nobody stepped up to maintain it. It's time to fully remove it.

This commit deletes all Consul logic, replaces Consul with Etcd in tests where it still was the default, removes Consul-specific tests, and removes the dependency to hashicorp/consul.

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
I think it deserves a note in upgrade notes (`Documentation/operations/upgrade.rst`)?

**Review by sayboras — APPROVED** [review]
I am in favor of this cleanup, let's see if others have something else in mind :thinking:

**Comment by bimmlerd** [comment]
Does this warrant a brief discussion in today's or next community meeting?

**Review by derailed — APPROVED** [review]
@pchaigno Always in favor of removing duds. Thanks for taking this on Paul!

**Comment by gandro** [comment]
Some users to notify might be found in https://github.com/cilium/cilium/issues/18851

File: .github/workflows/conformance-runtime.yaml [hunk 1]
```diff
@@ -188,8 +188,6 @@ jobs:
           # RuntimeAgentChaos Cilium agent removing leftover Cilium interfaces
           # RuntimeAgentChaos Connectivity over restarts Checking that during restart no traffic is dropped using Egress + Ingress Traffic
           # RuntimeAgentChaos Endpoint Endpoint recovery on restart
-          # RuntimeAgentChaos KVStore Delete event on KVStore with CIDR identities
-          # RuntimeAgentChaos KVStore Validate that delete events on KVStore do not release in use identities
```
Comment (aanm) [inline .github/workflows/conformance-runtime.yaml:190]: Aren't we losing coverage by removing these two tests?

Comment (pchaigno) [reply .github/workflows/conformance-runtime.yaml:190]: Despite not being clear in their names, these two tests were Consul-specific, with even a hardcoded Consul URL.

**Review by ghost — APPROVED** [review]
docs ok

**Review by tklauser — APPROVED** [review]

**Review by marseel — APPROVED** [review]
Thanks :broom:

**Review by marseel — CHANGES_REQUESTED** [review]
oh wait, I missed one directory to review... :facepalm:

File: pkg/kvstore/config.go [hunk 2]
```diff
@@ -5,15 +5,16 @@ package kvstore
...
 	"sync"
 
 	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
 var (
-	// selectedModule is the name of the selected backend module
-	selectedModule string
+	// ErrNotImplemented is the error which is returned when a functionality is not implemented.
+	ErrNotImplemented = errors.New("not implemented")
```
Comment (marseel) [inline pkg/kvstore/config.go]: nit: AFAIK this is only used in etcd_lease_test.go. I would move that error there now.

**Review by marseel — APPROVED** [review]

**Pull request merged**